Source/WebCore/dom/DocumentStorageAccess.cpp

```diff
@@ -198,6 +203,11 @@ void DocumentStorageAccess::requestStorageAccess(Ref<DeferredPromise>&& promise)
         return;
     }
 
+    if (!document->isSecureContext()) {
+        promise->reject(ExceptionCode::NotAllowedError);
+        return;
+    }
```

Comment:
Same here for `requestStorageAccessQuickCheck` (although maybe it won't reject with the same exception than the spec???, please double check)

Reply:
I can't here. I need the exception the spec specifies.